src/core/connection_pool.c

```diff
@@ -0,0 +1,651 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Definitions for the MsQuic Connection Pool API, which allows clients to
+    create a pool of connections that are spread across RSS cores.
+
+--*/
+
+#include "precomp.h"
+
+#ifdef QUIC_CLOG
+#include "connection_pool.c.clog.h"
+#endif
+
+#define MAX_CONNECTION_POOL_RETRY_MULTIPLIER 2
```

Comment:
Maybe we should do more...

Reply:
also, each success resets the retry counter, so it's actually ((2 * NumberOfConnections) - 1) * NumberOfConnections worst case number of retries.